x-pack/filebeat/input/httpjson/config_okta_auth.go

```diff
@@ -0,0 +1,192 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpjson
+
+import (
+	"context"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// customTokenSource is a custom implementation of the oauth2.TokenSource interface.
+// for more information, see https://pkg.go.dev/golang.org/x/oauth2#TokenSource
+type customTokenSource struct {
```

Comment:
In `customTokenSource`, "custom" is not all that descriptive. Perhaps `oktaTokenSource` would be more informative.